streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -4574,4 +4574,153 @@ object ZStream extends ZStreamPlatformSpecificConstructors {
     def single(a: A): B =
       apply(ZIO.succeedNow(Chunk(a)))
   }
+
+  implicit final class SortedByKey[R, E, K, A](private val self: ZStream[R, E, (K, A)]) {
```

Comment:
Is it possible to put this on the class itself with an implicit subtype evidence for the tuple? Should be more inline with the other operators.

Reply:
No unfortunately it leads to significantly worse type inference because then the input type to the combining function needs to be specified.

Reply:
Ah alright.